static/templates/compose_banner/compose_banner.hbs

```diff
@@ -9,7 +9,7 @@
     <div class="banner_content">{{> @partial-block}}</div>
     {{/if}}
     {{#if button_text}}
-    <button class="compose_banner_action_button" >{{button_text}}</button>
+    <button class="compose_banner_action_button{{#if hide_close_button}} right_edge{{/if}}" >{{button_text}}</button>
```

Comment:
Couldn't think of a particularly nice way to add the padding back in if the close button wasn't rendered -- open to suggestions!